jps/model-api/src/org/jetbrains/jps/model/module/JpsModuleSourceRootType.java

```diff
@@ -28,4 +28,9 @@
  * @author nik
  */
 public interface JpsModuleSourceRootType<P extends JpsElement> extends JpsElementType<P>, JpsElementTypeWithDefaultProperties<P> {
+
+  /**
+   * Returns true if SourceRoot represents test resources, false otherwise
+   */
+  boolean isForTests();
```

Comment:
This will break backward compatibility with plugins implementing this interface. `default` method should be used.